### PR TITLE
[libc] Remove _Exit proxy func header and use LIBC_NAMESPACE::_Exit in tests

### PR DESCRIPTION
**Pull request by ghost**
This improves/fixes this pr https://github.com/llvm/llvm-project/pull/114718. In this PR we removed the _Exit proxy func because it was not needed. Instead we used `LIBC_NAMESPACE::_Exit`

### CODE REVIEW

**Review by nickdesaulniers — APPROVED** [review]

**Review by michaelrj-google — CHANGES_REQUESTED** [review]
Why do we need a proxy header for `_Exit` at all? Why don't we just call our internal `_Exit` in the `atexit` and `atquickexit` tests?

**Comment by Jobhdez** [comment]
> Why do we need a proxy header for `_Exit` at all? Why don't we just call our internal `_Exit` in the `atexit` and `atquickexit` tests?

Yeah this makes sense. I will update the PR to remove the `_Exit` proxy header, and switch those 2 tests to use `LIBC_NAMESPACE::_Exit` instead. Can you please just confirm that this is what we’re going to do? Thanks

**Comment by michaelrj-google** [comment]
Yes that sounds like the correct thing to do. Thank you!

**Review by michaelrj-google — COMMENTED** [review]
LGTM, please update the title and description

**Comment by Jobhdez** [comment]
> LGTM, please update the title and description

ok I will. do you want me to merge after I change the title? or do you want to merge yourself?

**Review by michaelrj-google — APPROVED** [review]
This should be fine to merge as-is. I'd recommend waiting until the morning to merge it (so that people can help troubleshoot if necessary), but it's up to you.

**Comment by Jobhdez** [comment]
> This should be fine to merge as-is. I'd recommend waiting until the morning to merge it (so that people can help troubleshoot if necessary), but it's up to you.

ok ill wait until tomorrow morning. thanks. can you please suggest a new title for the pr? thanks

**Comment by lntue** [comment]
> > This should be fine to merge as-is. I'd recommend waiting until the morning to merge it (so that people can help troubleshoot if necessary), but it's up to you.
> 
> ok ill wait until tomorrow morning. thanks. can you please suggest a new title for the pr? thanks

You can change it to `[libc] Remove _Exit proxy func header and use LIBC_NAMESPACE::_Exit in tests.`, and update the description accordingly.  You might also want to remove @nickdesaulniers from the description, so that he won't be tagged every time the PR is merged in other forks.

**Comment by Jobhdez** [comment]
> > > This should be fine to merge as-is. I'd recommend waiting until the morning to merge it (so that people can help troubleshoot if necessary), but it's up to you.
> > 
> > 
> > ok ill wait until tomorrow morning. thanks. can you please suggest a new title for the pr? thanks
> 
> You can change it to `[libc] Remove _Exit proxy func header and use LIBC_NAMESPACE::_Exit in tests.`, and update the description accordingly. You might also want to remove @nickdesaulniers from the description, so that he won't be tagged every time the PR is merged in other forks.

ok sounds good. I'm going to merge now!

File: libc/test/src/stdlib/atexit_test.cpp
Comment (michaelrj-google) [inline libc/test/src/stdlib/atexit_test.cpp:9]: looks like you need to include `_Exit` here explicitly (`#include "src/stdlib/_Exit.h`). Applies to both tests.

**Review by michaelrj-google — COMMENTED** [review]
Tests are failing, I think I figured out why:

File: libc/test/src/stdlib/atexit_test.cpp
Comment (Jobhdez) [reply libc/test/src/stdlib/atexit_test.cpp:9]: ok will fix it right now

**Pull request merged**
